example/basic.go

```diff
@@ -0,0 +1,63 @@
+package main
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+
+	"github.com/RisingStack/trace-go/trace"
+)
+
+var collector = trace.NewMemoryCollector()
```

Comment:
I would have a "standard" collector in the package that is used by the `trace.XYZ` methods and functions we don't need to pass it around manually. And provide a method to override the default collector. Also provide a more finer grained approach - just like in this example.

I think the `log` and `http` packages are good examples for this. They have a sensible default as globally available and easy to use but you can have your own `Logger` instance.